raw_linux.go

```diff
@@ -78,8 +75,18 @@ func listenPacket(ifi *net.Interface, proto uint16, cfg Config) (*packetConn, er
 		return nil, err
 	}
 
+	if err := unix.SetNonblock(sock, true); err != nil {
+		return nil, err
+	}
+
+	f := os.NewFile(uintptr(sock), "sock")
```

Comment:
Can you leave a comment along the lines of what https://github.com/mdlayher/netlink/blob/master/conn_linux.go#L446-L451 says here too?  It's a subtle enough thing that I want to make sure it's adequately documented.

For the file name, I'd prefer something like "raw", "packet", etc rather than generic "sock".  Perhaps it makes sense to use "ethernet" when LinuxSockDGRAM is false, and "raw" when it's true or something.